Fix nfs external provisioner to use containerized backend instead of ganesha

mkimuram opened the pull request:
**What this PR does / why we need it**:
Existing nfs external provisioner's backend is ganesha and it is not suitable for large block size IO. So, replacing the backend to containerized nfs volume server.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #68097 

**Special notes for your reviewer**:
/sig storage

**Release note**:
```release-note
NONE
```

mkimuram commented:
@msau42 @wongma7 

PTAL

Should we choose a particular version of image, not latest?
Also, I hope that it is enough small change to merge in this time frame and code makes sense.

msau42 commented:
/ok-to-test

msau42 commented on test/e2e/storage/drivers/in_tree.go:
```diff
@@ -163,47 +166,39 @@ func startExternalProvisioner(c clientset.Interface, ns string, externalPluginNa
 				{
 					Name:  "nfs-provisioner",
-					Image: "quay.io/kubernetes_incubator/nfs-provisioner:v1.0.9",
+					Image: "quay.io/external_storage/nfs-client-provisioner:latest",
```
I think it would be better to use a specific version for the Kubernetes CI.

mkimuram replied:
@wongma7 

Which version do you recommend to use here? [v3.1.0-k8s1.11](https://quay.io/repository/external_storage/nfs-client-provisioner?tag=latest&tab=tags)?

mkimuram replied:
@msau42 

Changed the tag from latest to v3.1.0-k8s1.11, before getting answer from @wongma7 .
Because it should be better than latest and we can change it later if we find issue in v3.1.0-k8s1.11.

mkimuram committed:
Fix nfs external provisioner to use containerized backend instead of ganesha

Fixes: #68097

mkimuram commented:
@msau42 

I added a bit quick and dirty fix, to avoid volumes test failure by changing ganesha to containerized nfs server.

In volumes test, ```VolumeTestCleanup``` deletes server pod before dynamic provisioner is deleted, so deletion of dynamic provisioner fails.
To avoid it, I added a code to set ```""``` to ```config.ServerImage``` in ```nfsDriver.CreateDriver```.
Ideally, ```VolumeTestCleanup``` should be fixed not to delete nfs server pod, however, it is still called from many other places that require this behavior. 
So, it would be better to workaround this way for now, and do the actual fix in the future.


[volumes test log] (Without below code, nfs server is deleted before external-provisioner, which cause  external-provisoner deletion failure)
```
STEP: starting nfs client
STEP: Checking that text file contents are perfect.
Sep  4 14:43:19.716: INFO: Running '/root/go/src/k8s.io/kubernetes/_output/bin/kubectl --server=https://localhost:6443/ --kubeconfig=/var/run/kubernetes/admin.kubeconfig exec nfs-client --namespace=e2e-tests-volumes-8x9w6 -- cat /opt/0/index.html'
Sep  4 14:43:20.080: INFO: stderr: ""
Sep  4 14:43:20.080: INFO: stdout: "Hello from nfs from namespace e2e-tests-volumes-8x9w6\n"
STEP: cleaning the environment after nfs
Sep  4 14:43:20.084: INFO: Deleting pod "nfs-client" in namespace "e2e-tests-volumes-8x9w6"
Sep  4 14:43:20.091: INFO: Wait up to 5m0s for pod "nfs-client" to be fully deleted
Sep  4 14:43:36.133: INFO: Deleting pod "nfs-server" in namespace "e2e-tests-volumes-8x9w6"
Sep  4 14:43:36.139: INFO: Wait up to 5m0s for pod "nfs-server" to be fully deleted
[AfterEach] [Testpattern: Dynamic PV (default fs)] volumes
  /root/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/testsuites/volumes.go:132
[AfterEach] [Driver: nfs]
  /root/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/storage/in_tree_volumes.go:83
Sep  4 14:43:44.226: INFO: Deleting pod "external-provisioner-ksqts" in namespace "e2e-tests-volumes-8x9w6"
Sep  4 14:43:44.232: INFO: Wait up to 5m0s for pod "external-provisioner-ksqts" to be fully deleted
```

[Code added]
```
--- a/test/e2e/storage/drivers/in_tree.go
+++ b/test/e2e/storage/drivers/in_tree.go
@@ -134,6 +134,11 @@ func (n *nfsDriver) CreateDriver() {
        n.externalPluginName = fmt.Sprintf("example.com/nfs-%s", ns.Name)
 
        n.driverInfo.Config, n.serverPod, n.serverIP = framework.NewNFSServer(cs, ns.Name, []string{})
+       // TODO(mkimuram): Below code is needed for making volumes test succeed by
+       // avoiding NFS server from deleted in VolumeTestCleanup before dynamic provisioner pod is deleted.
+       // VolumeTestCleanup should be fixed not to delete NFS server, after all volumes tests are changed
+       // to be called from this framework.
+       n.driverInfo.Config.ServerImage = ""
```

mkimuram commented:
@msau42 

With above fix added, all test passed properly now. PTAL.

msau42 commented:
lgtm but it will conflict with some refactoring being done in https://github.com/kubernetes/kubernetes/pull/68138

msau42 commented:
@mkimuram I don't think #68138 will resolve the issue.  It only fixes tests running using containerd.  I think it would still be better to change to nfs-client.  @wongma7 wdyt?
